components/CreateNftPage/CreateDropModal/index.tsx

```diff
@@ -0,0 +1,251 @@
+}
+
+const CreateDropModal = ({ setIsCreateDropModal }: IProps) => {
+  const [previewUrl, setPreviewUrl] = useState("");
+  const [newDropIsLoading, setNewDropImageIsLoading] = useState(false);
+  const [fileSavingFailed, setFileSavingFailed] = useState(false);
+  const [errorwhileSendingTransaction, setErrorWhileSendingTransaction] =
+    useState(false);
+  const [createDropProcessing, setCreateDropProcessing] = useState(false);
+  const [dropSavingOnDatabaseFailed, setDropSavingOnDatabasefailed] =
+    useState(false);
+  const [sendStorageFeeProcessing, setSendStorageFeeProcessing] =
+    useState(false);
+  const [isCreateDropProcessingModal, setIsCreateDropProcessingModal] =
+    useState(false);
+
+  const [dropData, setDropData] = useState<ICreateDrop>({
+    creationFeeTransactionHash: "",
+    creatorId: 0,
+    creatorAddress: "",
+    creatorUsername: "",
+    description: "",
+    dropID: "",
+    imageUrl: "",
+    imageUrlThumbnail: "",
+    title: "",
+    collection: true,
+  });
+
+  const userAccount = useRecoilValue(currentAccountState);
+
+  const inputFileRef = useRef<HTMLInputElement>(null);
+
+  const onCloseCreateDropModal = () => {
+    setIsCreateDropModal(false);
+    setIsCreateDropProcessingModal(false);
+  };
+
+  const onChooseFile = () => {
+    inputFileRef.current?.click();
+  };
+
+  const onFileChange = async (event) => {
+    const { files } = event.target;
+
+    setNewDropImageIsLoading(true);
+    const imageUrl = await saveFileWithWeb3Storage(files);
+    setNewDropImageIsLoading(false);
+
+    if (!imageUrl) {
+      setFileSavingFailed(true);
+      return;
+    }
+
+    setFileSavingFailed(false);
+
+    const previewUrl = URL.createObjectURL(files[0]);
+    setPreviewUrl(previewUrl);
+
+    setDropData({ ...dropData, imageUrl, imageUrlThumbnail: imageUrl });
+  };
+
+  const onDropDetailsChange = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    const { value, name } = event.target;
+
+    setDropData({ ...dropData, [name]: value });
+  };
+
+  const onCreateDrop = async () => {
+    if (
+      !dropData.description ||
+      !dropData.title ||
+      !dropData.imageUrl ||
+      !dropData.imageUrlThumbnail
+    )
+      return;
+
+    setIsCreateDropProcessingModal(true);
+    const web3Services = new Web3Service();
+
+    setSendStorageFeeProcessing(true);
+    const { transactionHash } = await web3Services.sendStorageFee();
+    setSendStorageFeeProcessing(false);
+
+    if (!transactionHash) {
+      setErrorWhileSendingTransaction(true);
+      return;
+    }
+
+    setErrorWhileSendingTransaction(false);
+
+    const drop: ICreateDrop = {
+      ...dropData,
+      creationFeeTransactionHash: transactionHash,
+      creatorAddress: `${
+        userAccount?.walletAddress ||
+        LocalStorage.getItem("ongama_signer_address")
+      }`,
+      creatorId: Number(userAccount?.id),
+      creatorUsername: `${userAccount?.username}`,
+      dropID: uuid(),
```

Comment:
but it's not handled on backend actually, and it's required in `createDropDto` in swagger, that's why I used UUID to generate a dropID

Reply:
btw I removed the UUID package

Reply:
It shouldn't be required on the front-end.

You can reach out to the backend team to remove it.

It may be required in cases where you have first to save the drop to the smart contract then from the result emitted by the event you grab the `dropID` and send it to the backend.

You should confirm with the backend guys about the flow.